### PR TITLE
added Hair fall prediction using machine learning

### LINKED ISSUE #799: 💡[Feature]:  HairFall detection using machine learning
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

i want to  make a system which can predict the hair fall in the person using machine learning models. i twill take various parameters as input like stress level, keratin level , hair texture , vitamin and manganese level ,hormonal changes ,genetic medical condition ,medical treatment ,stress , age , poor hair care , environmental factors etc

kaggle dataset link :https://www.kaggle.com/datasets/amitvkulkarni/hair-health (csv)
                                 https://www.kaggle.com/datasets?search=hair+loss  (images)
                                 
                                 i will develop an application using streamlit with an option to detect hairfall from your image as well as inputted parameters (both ml and deep learning models)
                                 
will do a complete data analysis , eda ,feature engineering etc 

### Use Case

This can be used to detect hair fall or any hair disease at early stages

### Benefits

this will reduce the problem  of hair fall

### Add ScreenShots

none

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by RajKhanke**
## Description

This project involves developing a Streamlit web application for predicting hair loss using both input-based and image-based approaches. The input-based prediction utilizes a pre-trained Naive Bayes model to predict hair loss based on user-provided information such as genetics, hormonal changes, medical conditions, medications, nutritional deficiencies, stress levels, age, poor hair care habits, environmental factors, smoking, and weight loss. The image-based prediction employs a pre-trained CNN model to classify uploaded images as either "Hair" or "No Hair" by resizing and preprocessing the image before making predictions. The app is designed with an intuitive and user-friendly interface, featuring custom CSS for enhanced styling and a sidebar for navigation. Users can either fill out a form with their personal and health details or upload an image to receive a prediction about their hair loss status. The application also maps the model's numerical predictions back to human-readable labels and provides appropriate success or error messages based on the results.

 
-  CSV Model :
The five machine learning model random forest ,xgboost ,KNN, gradient boosting machine, Naive Bayes are selected for model training over the inputed processed data.
   
The naive bayes machine model is then loaded into streamlit application after installing and using joblib library and performing 10 fold cross validation over it.

- image classification model :
CNN model of deep learning is used for image classification which goes through various steps of data analysis, processing, feature engineering , scaling and data augmentation on batch size of 32.

https://github.com/user-attachments/assets/87a92edb-f123-417f-968c-ee5d518fb40b

Fixes #799 

## Type of change

- [X] Added a new machine learning frameworks, libraries or software.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

**Comment by RajKhanke** [comment]
@sanjay-kv please merge this pr along with the level and label of gssoc'24. it will resolve issue no 799. also this is my last contribution to this repo as per max point rules and regulations
